File: Examples/Node.js/SDK/PivotTables/AddPivotFieldInPivottable.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var StorageApi = require('asposestoragecloud');
var CellsApi = require('asposecellscloud');
var configProps = require('../Config/config.json');
var data_path = '../../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose Cells API SDK
var cellsApi = new CellsApi(config);

// Set input file name
var name = "Sample_Pivot_Table_Example.xls";
var sheetName = "Sheet2";
var pivotTableIndex = 0;
var pivotFieldType = "Row";
var pivotTableFieldRequest = {
		'Data' : [1,2]
};

try {
// Upload source file to aspose cloud storage
	storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {
	assert.equal(responseMessage.status, 'OK');
	
	// Invoke Aspose.Cells Cloud SDK API to add pivot field into pivot table
	cellsApi.PutPivotTableField(name, sheetName, pivotTableIndex, pivotFieldType, null, null, pivotTableFieldRequest, function(responseMessage) {
		assert.equal(responseMessage.status, 'OK');
		
		// Download updated Workbook from storage server
		storageApi.GetDownload(name, null, null, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			var outfilename = name;
			var writeStream = fs.createWriteStream(data_path + "_out_" + outfilename);
			writeStream.write(responseMessage.body);
			});
		});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Python/AddPivotFieldInPivotTable.py
mport asposecellscloud
from asposecellscloud.CellsApi import CellsApi
from asposecellscloud.CellsApi import ApiException
from asposecellscloud.models import PivotTableFieldRequest

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Cells API SDK
api_client = asposecellscloud.ApiClient.ApiClient(apiKey, appSid, True)
cellsApi = CellsApi(api_client);

#set input file name
filename = "Sample_Pivot_Table_Example.xls"
sheetName = "Sheet2"
pivotTableIndex = 0
pivotFieldType = "Row"

body = PivotTableFieldRequest.PivotTableFieldRequest()
body.Data = [1,2]

#upload file to aspose cloud storage
storageApi.PutCreate(Path=filename, file=data_folder + filename)

try:
    #invoke Aspose.Cells Cloud SDK API to add pivot field into pivot table
    response = cellsApi.PutPivotTableField(name=filename, sheetName=sheetName, pivotTableIndex=pivotTableIndex, pivotFieldType=pivotFieldType, body=body)

    if response.Status == "OK":
        #download Workbook from storage server
        response = storageApi.GetDownload(Path=filename)
        outfilename = "c:/temp/" + filename
        with open(outfilename, 'wb') as f:
                    for chunk in response.InputStream:
                        f.write(chunk)

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
